### PR TITLE
feat: add extra theme colors

### PR DESCRIPTION
**Pull request by el**
### Describe what this PR does
- Adds more theme colors to the banner

cabbage - onion (*fixed)
Screen Shot 2022-10-14 at 15 34 39

cabbage:
Screen Shot 2022-10-14 at 12 24 54

bacon
Screen Shot 2022-10-14 at 12 22 43

ketchup
Screen Shot 2022-10-14 at 12 23 27

bun:
Screen Shot 2022-10-14 at 12 25 37

cheese:
Screen Shot 2022-10-14 at 12 26 53

lettuce:
Screen Shot 2022-10-14 at 12 27 11

avocado 
Screen Shot 2022-10-14 at 12 20 04

blue cheese:
Screen Shot 2022-10-14 at 12 27 38

burger:
Screen Shot 2022-10-14 at 12 28 41

white/pepper90:
Screen Shot 2022-10-14 at 12 29 12

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [x] Does this not break anything in companion?

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-669 #done

### CODE REVIEW

File: packages/shared/src/components/PromotionalBanner.tsx [hunk 1]
```diff
@@ -3,12 +3,28 @@ import classNames from 'classnames';
...
+import { BannerCustomTheme, BannerData, BannerTheme } from '../graphql/banner';
+import { Theme } from './utilities';
 
 interface PromotionalBannerProps {
   bannerData: BannerData;
   setLastSeen: (value: Date) => Promise<void>;
 }
+
+const classNamesByTheme: Record<BannerTheme, string[]> = {
+  [BannerCustomTheme.cabbageOnion]: ['bg-theme-color-cabbage bg-gradient-to-b', 'text-white', 'bg-white text-pepper-90'],
```
Comment (el) [inline packages/shared/src/components/PromotionalBanner.tsx:15]: @rebelchris is `bg-gradient-to-b` correct for a gradient bottom?

File: packages/shared/src/graphql/banner.ts [hunk 2]
```diff
@@ -1,10 +1,18 @@
 import { gql } from 'graphql-request';
+import { Theme } from '../components/utilities';
+
+export enum BannerCustomTheme {
+  cabbageOnion = 'cabbage-onion',
+  whitePepper = 'whitePepper',
+}
+
+export type BannerTheme = Theme | BannerCustomTheme;
```
Comment (el) [inline packages/shared/src/graphql/banner.ts:9]: AFAIK we dont have a shared type system between the backend and frontend right?

File: packages/shared/src/components/PromotionalBanner.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/PromotionalBanner.tsx:15]: Yep, though these ones are not yet whitelisted from the rule if I'm not mistaken. We can add this manually here:
https://github.com/dailydotdev/apps/blob/master/packages/eslint-rules/lib/rules/no-custom-color.js#L13

I've seen multiple introductions to new classes like this. In the future, probably would need to make our custom rule accept a prefix and automatically whitelist everything that follows.
Something like `bg-gradient-` will white list:
`bg-gradient-to-b`,
`bg-gradient-to-br`,
etc.

File: packages/shared/src/graphql/banner.ts (hunk 2 above)
Comment (idoshamun) [reply packages/shared/src/graphql/banner.ts:9]: You're right

**Review by rebelchris — APPROVED** [review]
No comments on the code part, looks solid, thanks for taking this on 🙏

I do want to raise a test scenario.
What happens for people who don't have this version (extension users) when we set the alert to one of these new ones?

So it's unrelated to this code, but a good check to do.

File: packages/shared/src/components/utilities.tsx [hunk 3]
```diff
@@ -4,6 +4,18 @@ import styles from './utilities.module.css';
...
+export enum Theme {
+  avocado = 'avocado',
+  bacon = 'bacon',
+  blueCheese = 'blue-cheese',
+  bun = 'bun',
+  burger = 'burger',
+  cabbage = 'cabbage',
+  cheese = 'cheese',
+  ketchup = 'ketchup',
+  lettuce = 'lettuce',
```
[the remark this answers is not in the document]
Comment (el) [reply packages/shared/src/components/utilities.tsx:16]: My bad, looks like PascalCase is being used on enums. BTW we can add this as a linter rule

File: packages/shared/src/components/PromotionalBanner.tsx [hunk 4]
```diff
@@ -3,12 +3,29 @@ import classNames from 'classnames';
...
 
 interface PromotionalBannerProps {
   bannerData: BannerData;
   setLastSeen: (value: Date) => Promise<void>;
 }
+
+const classNamesByTheme: Record<BannerTheme, string[]> = {
+  [BannerCustomTheme.cabbageOnion]: [
+    'from-theme-color-cabbage to-theme-color-onion bg-gradient-to-r', 'text-white', 'bg-white text-pepper-90'],
+  [BannerCustomTheme.whitePepper]: ['bg-white', 'text-pepper-90', 'bg-theme-color-cabbage'],
```
[the remark this answers is not in the document]
Comment (el) [reply packages/shared/src/components/PromotionalBanner.tsx]: Nice catch. here is the fixed version:
Screen Shot 2022-10-14 at 17 08 14
Screen Shot 2022-10-14 at 17 08 27

**Pull request merged**